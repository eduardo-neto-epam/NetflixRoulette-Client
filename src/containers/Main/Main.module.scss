.background {
    background: var(--color-dark);
    width: 100vw;
    flex-grow: 2;
    position: relative;
    margin-top: 20px;

    .container {
        width: var(--content-max-width);
        position: absolute;
        top: 0;
        left: 50%;
        transform: translateX(-50%);
        height: 100%;

        .menu-row {
            min-width: 100%;
            display: flex;
            justify-content: space-between;
            align-items: center;
            position: relative;
        }

        .separator {
            border-top: 3px double var(--color-medium);
            min-width: 100%;
        }

        .inner {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 50%;
            
            
            .no-movies-warn {
                display: inline-block;
                width: 1000px;
                margin: 40px auto;
                color: var(--color-contrast);
                font-family: var(--font-main);
                font-weight: var(--weight-light);
                font-size: 48px;
                line-height: 150%;
                letter-spacing: 3px;
                text-align: center;
            }
        }
    }
}

