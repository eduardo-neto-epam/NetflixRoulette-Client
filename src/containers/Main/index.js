import React from 'react'
import styles from './Main.module.scss'
import classnames from 'classnames/bind'

import FiltersMenu from '../FiltersMenu'
import SortSelector from '../../components/SortSelector'
import Footer from '../Footer'

let cx = classnames.bind(styles)

const Main = () => {
    return (
        <div className={cx('background')}>
            <div className={cx('container')}>
                <div className={cx('menu-row')}>
                    <FiltersMenu />
                    <SortSelector />
                </div>
                <hr className={cx('separator')}/>
                <div className={cx('inner')}>
                    <h3 className={cx('no-movies-warn')}>No Movie Found</h3>
                </div>
            </div>
            <Footer />
        </div>
    )
}

export default Main