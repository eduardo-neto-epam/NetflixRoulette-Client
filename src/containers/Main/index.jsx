import React, { useEffect, useState } from 'react';
import classnames from 'classnames/bind';
import styles from './Main.module.scss';
import { sortMovies } from '../../utils/sortFunctions/index.js';
import filterMovies from '../../utils/filterMovies/index.js';

import FiltersMenu from '../FiltersMenu/index.jsx';
import SortSelector from '../../components/SortSelector/index.jsx';
import Footer from '../Footer/index.jsx';
import Counter from '../../components/Counter/index.jsx';
import ErrorBoundary from '../../components/ErrorBoundary/index.jsx';
import CardMovie from '../CardMovie/index.jsx';

import DummyData from '../../dummyData/index.json';

const cx = classnames.bind(styles);

const Main = () => {
  const [movies, setMovies] = useState([]);
  const [filter, setFilter] = useState('all');
  const filters = [
    'all',
    'documentary',
    'comedy',
    'horror',
    'crime',
    'action',
  ];
  const [selectedMethod, setSelectedMethod] = useState('release date');
  const sortOptions = [
    'release date',
    'movie title',
    'movie genre',
  ];

  useEffect(() => {
    const data = DummyData;
    const sortedData = sortMovies(data, selectedMethod, filter);
    setMovies(() => [...sortedData]);
  }, [selectedMethod]);

  useEffect(() => {
    const data = DummyData;
    const filteredMovies = filterMovies(filter, data);
    setMovies(() => [...filteredMovies]);
    // eslint-disable-next-line no-console
    console.log('Current Filter is: ', filter);
  }, [filter]);

  return (
    <div className={cx('background')}>
      <div className={cx('container')}>
        <div className={cx('menu-row')}>
          <FiltersMenu filter={filter} setFilter={setFilter} filters={filters} />
          <SortSelector
            options={sortOptions}
            selectedMethod={selectedMethod}
            setSelectedMethod={setSelectedMethod}
          />
        </div>
        <hr className={cx('separator')} />
        <ErrorBoundary>
          <div className={cx({ inner: !movies.length }, { 'inner--display': movies.length })}>
            { !movies.length && <h3 className={cx('no-movies-warn')}>No Movie Found</h3>}
            { movies.length > 0 && <Counter count={movies.length} />}
            { movies.length > 0 && (
            <div className={cx('wrapper')}>
              { movies.length > 0 && movies.map((movie) => (
                <CardMovie
                  key={movie.id}
                  movieID={movie.id}
                  imageSrc={movie.poster_path}
                  imageAlt={`Poster for movie ${movie.title}`}
                  movieTittle={movie.title}
                  movieReleaseDate={movie.release_date}
                  movieGenres={movie.genres}
                  movieOverview={movie.overview}
                  movieRuntime={movie.runtime}
                />
              )) }
            </div>
            ) }
          </div>
        </ErrorBoundary>
      </div>
      <Footer />
    </div>
  );
};

export default Main;
