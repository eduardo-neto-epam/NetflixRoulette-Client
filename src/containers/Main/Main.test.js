import React from 'react'
import '@testing-library/react'
import renderer from 'react-test-renderer';
import Main from './index'

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

test('Main renders correctly', () => {
    const tree = renderer
        .create(<Main />)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });