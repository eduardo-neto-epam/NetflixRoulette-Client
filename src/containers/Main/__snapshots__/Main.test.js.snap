// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<div
  className="background"
>
  <div
    className="container"
  >
    <div
      className="menu-row"
    >
      <ul
        className="menu"
      >
        <li
          className="filter"
        >
          <label
            className="filter"
            htmlFor="all"
          >
            all
            <input
              className="menu-filter"
              defaultChecked={true}
              id="all"
              name="filters"
              onChange={[Function]}
              type="radio"
              value="all"
            />
            <span
              className="checkmark"
            />
          </label>
        </li>
        <li
          className="filter"
        >
          <label
            className="filter"
            htmlFor="documentary"
          >
            documentary
            <input
              className="menu-filter"
              defaultChecked={false}
              id="documentary"
              name="filters"
              onChange={[Function]}
              type="radio"
              value="documentary"
            />
            <span
              className="checkmark"
            />
          </label>
        </li>
        <li
          className="filter"
        >
          <label
            className="filter"
            htmlFor="comedy"
          >
            comedy
            <input
              className="menu-filter"
              defaultChecked={false}
              id="comedy"
              name="filters"
              onChange={[Function]}
              type="radio"
              value="comedy"
            />
            <span
              className="checkmark"
            />
          </label>
        </li>
        <li
          className="filter"
        >
          <label
            className="filter"
            htmlFor="horror"
          >
            horror
            <input
              className="menu-filter"
              defaultChecked={false}
              id="horror"
              name="filters"
              onChange={[Function]}
              type="radio"
              value="horror"
            />
            <span
              className="checkmark"
            />
          </label>
        </li>
        <li
          className="filter"
        >
          <label
            className="filter"
            htmlFor="crime"
          >
            crime
            <input
              className="menu-filter"
              defaultChecked={false}
              id="crime"
              name="filters"
              onChange={[Function]}
              type="radio"
              value="crime"
            />
            <span
              className="checkmark"
            />
          </label>
        </li>
      </ul>
      <div
        className="container"
      >
        <div
          className="_inner"
        >
          <div
            className="label"
          >
            SORT BY
          </div>
          <div
            className="selected-method"
            data-testid="selected-method"
            onClick={[Function]}
          >
            release date
          </div>
        </div>
        <ul
          className="options-container"
        >
          <li
            className="option"
            onClick={[Function]}
          >
            release date
          </li>
          <li
            className="option"
            onClick={[Function]}
          >
            movie title
          </li>
          <li
            className="option"
            onClick={[Function]}
          >
            movie genre
          </li>
        </ul>
      </div>
    </div>
    <hr
      className="separator"
    />
    <div
      className="inner"
    >
      <h3
        className="no-movies-warn"
      >
        No Movie Found
      </h3>
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      <span
        className="brand brand--bold"
      >
        netflix
      </span>
      <span
        className="brand"
      >
        roulette
      </span>
    </div>
  </div>
</div>
`;
